test/test_indexing.py

```diff
@@ -1328,6 +1328,17 @@ def test_gather_take_along_dim_cross_device(self, device, dtype):
                                     r"Expected tensor to have .* but got tensor with .* torch.take_along_dim()"):
             torch.take_along_dim(t.cpu(), indices, dim=0)
 
+    @onlyCUDA
+    def test_cuda_broadcast_index_use_deterministic_algorithms(self, device):
+        torch.use_deterministic_algorithms(True)
+        tensor_a = torch.rand(100, 4).cpu()
+        tensor_b = tensor_a.to(device=device)
+        idx = torch.tensor([0, 1])
+        tensor_a[idx, :] = 1.0
+        tensor_b[idx, :] = 1.0
```

Comment:
Could you add some more test cases? I'd like to make sure it works for indices with multiple colons in different dimensions, like `tensor[:, idx_dim1, :, idx_dim3]`, for instance. It would be good to have a few cases with colons in different places. Also, we should probably have a test for the case you mentioned here: https://github.com/pytorch/pytorch/issues/105819#issue-1817542881

Reply:
Thank you for your advice, I have fixed it!